File: presets/java/src/test/java/code/retreat/java/CellTest.java
package code.retreat.java;

import static org.junit.Assert.assertTrue;

import org.junit.Test;

public class CellTest {


    @Test
    public void a_new_cell_is_in_dead_state() throws Exception {
        var cell = new Cell();

        assertTrue(cell.isDead());
    }
    
    @Test
    public void a_cell_with_2_neighbours_evolves_to_alive() throws Exception {
        var cell = new Cell();
        cell.evolveWith(2);

        assertTrue(cell.isAlive());
    }

}

File: presets/scala/src/test/scala/code/retreat/scala/CellTest.scala
package code.retreat.scala

import org.scalatest._
import org.scalatest.Matchers._

class CellTest extends WordSpec {
  "Cell" should {
    "Return not dead if the Cell is alive" in {
      val cell = Cell(isAlive = true)
      cell.isAlive should not be cell.isDead
    }
  }
}
